benchmark/runner.py

```diff
@@ -99,13 +100,18 @@ def run(definition, dataset, count, run_count, rebuild):
             descriptor["index_size"] = index_size
             descriptor["algo"] = definition.algorithm
             descriptor["dataset"] = dataset
-            store_results(dataset, count, definition, query_arguments,
-                    descriptor, results, search_type)
+
+            if power_capture.enabled():
+                power_stats = power_capture.run(algo, X, distance, count,
+                                run_count, search_type, descriptor)
+
+            store_results(dataset, count, definition,
+                    query_arguments, descriptor, results, search_type) 
     finally:
         algo.done()
 
 
-def run_from_cmdline():
+def run_from_cmdline(args=None):
```

Comment:
I'm not sure if this is very clever or very dangerous in the long run :-)

Reply:
Probably more the latter :)